server/src/main/java/org/elasticsearch/index/query/SpanTermQueryBuilder.java

```diff
@@ -78,7 +83,25 @@ protected SpanQuery doToQuery(SearchExecutionContext context) throws IOException
             term = new Term(fieldName, BytesRefs.toBytesRef(value));
         } else {
             Query termQuery = mapper.termQuery(value, context);
-            term = MappedFieldType.extractTerm(termQuery);
+            List<Term> termsList = new ArrayList<>();
+            termQuery.visit(new QueryVisitor() {
+                @Override
+                public QueryVisitor getSubVisitor(BooleanClause.Occur occur, Query parent) {
```

Comment:
Why do we need this override this way? Is it because of potentially having to unwrap BoostQuery around a TermQuery? I'm trying to understand why we only descend on MUST and FILTER clauses.

Reply:
good catch. This is to reproduce the previous boost query unwrapping. I am trying to figure out how to repro that so that I can add a unit test. Do you have ideas? When would we get a BoostQuery or anything else that wraps a term query out of calling mapper.termQuery ?  if it was a problem for boost query, why wasn't it for e.g. ConstantScoreQuery? Either we were not covering enough queries with the previous instance of checks, or the check for boost query was not needed, or? :)

Reply:
I thought having a query boost would wrap every other query? At least that's what AbstractQueryBuilder#toQuery() looks like, so setting a boost on the test query should work I think.

Reply:
right, I see your memories of the query refactoring are still alive, mine are gone :) I added a test around that, at least we know it worked before and it still works now. The query visitor is much more general which is probably overkill, but I am reluctant to check query types in there, I think the check for number of terms at the end provides the safety we need.